validator/src/main.rs

```diff
@@ -1978,7 +1984,17 @@ pub fn main() {
 
     let private_rpc = matches.is_present("private_rpc");
     let no_port_check = matches.is_present("no_port_check");
-    let no_rocksdb_compaction = matches.is_present("no_rocksdb_compaction");
+    let rocksdb_compaction = match (
+        matches.is_present("rocksdb_compaction"),
+        matches.is_present("no_rocksdb_compaction"),
+    ) {
+        (true, true) => panic!("conflicting options are supplied!"),
+        (true, false) => true,
+        (false, true) => false,
+        (false, false) => false, // no MANUAL compaction as default behavior
```

Comment:
I changed default behavior for easier testing. I'll revert this before merging this after it so that we can relatively safely merge this pr and backport to v1.6 (effectively only add new options)

Reply:
@carllin well, I changed my mind regarding this (cd4a5b0cbb80e3760e2b82bd1868625182142f2c). I think changing the default behavior straightly is actually enough. What do you think?

- Let's just make the default behavior work best. Avoid yet another flag to bother validators.
- Plumbing and maintaining and testing combination of these options are tedious.
- There is no sensible usecase for reverting to old behavior because I'm very confident existing behavior is suboptimal after these various experiments.
- the default value (1 day) for `periodic_compaction_seconds` should be fairly reasonable and no need to change it.
- If we bricked ledger by new behavior code, reviving old compaction implementation also doesn't fix the ledger miraclely. we must bite the bullets.

As for release plan, I'm thinking like this:

1. merge this after LGTM
2. wait testnet to go v1.7
3. back port this onto v1.7.0 or v1.7.1 (maybe 1.7.1 is better to make 1.6 => 1.7 less riskier)
4. wait and see one or two patch releases. (if there is a real demand, I can add flags at this point....)
5. backport this onto v1.6.x for mainnet-beta.

Reply:
@carllin ping: ^^

Reply:
hmmm yeah, it doesn't seem like there's any obvious compatibility issues, so as long as it bakes on testnet and we do a slow rollout to a few 1.6.x nodes at a time, I think it should be ok. 

Side note, I think this means that this case is no longer ever hit? : https://github.com/solana-labs/solana/pull/16697/files#diff-317f52baf65751e39be1d7e1cbe5cefd2738f2828dd79293032de1d971d425b3L175-L181

Reply:
> Side note, I think this means that this case is no longer ever hit? : https://github.com/solana-labs/solana/pull/16697/files#diff-317f52baf65751e39be1d7e1cbe5cefd2738f2828dd79293032de1d971d425b3L175-L181

You mean this arm for `PurgeType::PrimaryIndex`? https://github.com/solana-labs/solana/pull/16697/files#r629787232

In that case, yeah, but I kept it to make this pr less obtrusive and, to make testing easy, and to make the flag addition future pr easy (if ever needed)